lib/postgres.go

```diff
@@ -0,0 +1,1414 @@
+			// TODO: Block height?
+			UtxoType: utxo.OutputType,
+			isSpent:  utxo.Spent,
+			UtxoKey: &UtxoKey{
+				TxID:  *utxo.OutputHash,
+				Index: utxo.OutputIndex,
+			},
+		})
+
+	}
+
+	return utxoEntries
+}
+
+//
+// BlockView Flushing
+//
+
+func (postgres *Postgres) FlushView(view *UtxoView) error {
+	return postgres.db.RunInTransaction(postgres.db.Context(), func(tx *pg.Tx) error {
+		if err := postgres.flushUtxos(tx, view); err != nil {
+			return err
+		}
+		if err := postgres.flushProfiles(tx, view); err != nil {
+			return err
+		}
+		if err := postgres.flushPosts(tx, view); err != nil {
+			return err
+		}
+		if err := postgres.flushLikes(tx, view); err != nil {
+			return err
+		}
+		if err := postgres.flushFollows(tx, view); err != nil {
+			return err
+		}
+		if err := postgres.flushDiamonds(tx, view); err != nil {
+			return err
+		}
+		if err := postgres.flushMessages(tx, view); err != nil {
+			return err
+		}
+		if err := postgres.flushCreatorCoinBalances(tx, view); err != nil {
+			return err
+		}
+
+		return nil
+	})
+}
+
+func (postgres *Postgres) flushUtxos(tx *pg.Tx, view *UtxoView) error {
+	var outputs []*PGTransactionOutput
+	for utxoKeyIter, utxoEntry := range view.UtxoKeyToUtxoEntry {
+		// Making a copy of the iterator is required
+		utxoKey := utxoKeyIter
+		outputs = append(outputs, &PGTransactionOutput{
+			OutputHash:  &utxoKey.TxID,
+			OutputIndex: utxoKey.Index,
+			OutputType:  utxoEntry.UtxoType,
+			PublicKey:   utxoEntry.PublicKey,
+			AmountNanos: utxoEntry.AmountNanos,
+			Spent:       utxoEntry.isSpent,
+		})
+	}
+
+	_, err := tx.Model(&outputs).WherePK().OnConflict("(output_hash, output_index) DO UPDATE").Insert()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (postgres *Postgres) flushProfiles(tx *pg.Tx, view *UtxoView) error {
+	var insertProfiles []*PGProfile
+	var deleteProfiles []*PGProfile
+	for pkidIter, profileEntry := range view.ProfilePKIDToProfileEntry {
+		// We don't have to do actually delete these profile entries. The DO UPDATE clause takes care of these changes.
+		if profileEntry.isDeleted {
+			glog.Infof("DELETED PROFILE %v %v", profileEntry.PublicKey, profileEntry.Username)
+			continue
+		}
+
+		// Making a copy of the iterator is required
+		pkid := pkidIter
+		profile := &PGProfile{
+			PKID:                    &pkid,
+			PublicKey:               NewPublicKey(profileEntry.PublicKey),
+			Username:                string(profileEntry.Username),
+			Description:             string(profileEntry.Description),
+			ProfilePic:              profileEntry.ProfilePic,
+			CreatorBasisPoints:      profileEntry.CreatorBasisPoints,
+			BitCloutLockedNanos:     profileEntry.BitCloutLockedNanos,
+			NumberOfHolders:         profileEntry.NumberOfHolders,
+			CoinsInCirculationNanos: profileEntry.CoinsInCirculationNanos,
+			CoinWatermarkNanos:      profileEntry.CoinWatermarkNanos,
+		}
+
+		if profileEntry.isDeleted {
+			deleteProfiles = append(deleteProfiles, profile)
```

Comment:
I don't think this code would ever hit because you're continueing above

Reply:
Thank you! The continue code needed to be deleted.